src/writer.js

```diff
@@ -14,13 +14,15 @@ function hasFiles (configuration) {
 function ensurePath (dir, fullPath) {
   return new Promise((resolve, reject) => {
     if (!fs.existsSync(dir)) {
-      mkdirp(dir, (err) => {
-        if (err) {
-          reject(new Error(`Failed to create path '${dir}' for configuration file '${fullPath}':\n\t${err.message}`))
-        } else {
-          resolve()
-        }
-      })
+      mkdirp(dir).then(
+        () => { resolve() },
```

Comment:
Can you just pass the `resolve` function itself here? 🤔

Reply:
@lumaxis - I'm not sure - can you elaborate? (Sorry - my foo.js is weak 😂 )

Reply:
```js
mkdirp(dir).then(resolve, (err) => {})
```

Reply:
I.e. instead of calling `resolve()` you can just pass a reference to `resolve` function.

Reply:
Cool - thanks! I'll try this (after I've got the tests working at all locally again :/ )

Reply:
@owenniblock interrupting just for the js syntax -- is there a test that will ensure that we get into that error block? the syntax in this file doesn't look like js promises, but i don't want to suggest a change from that perspective, just from "is there a test that ensures we get to this block, and is it passing" and let the code be what it needs to.

Reply:
I believe I have all the lines covered by tests now so hopefully we've covered all eventualities.